### PR TITLE
[Feat] Add alpha and color support to FOW

### PR DESCRIPTION
**Pull request by Bjorgenas**
**Marks**  https://github.com/otfbm/issues/issues/9

Passes FG mode color and grid opacity to FogEffect in Board

### CODE REVIEW

**Comment by someeric** [comment]
Still WIP regarding adding test cases.

**Comment by someeric** [comment]
**Ready for review.**  Tested locally and unit tests passing.

**Comment by someeric** [comment]
Just found out about this: https://github.com/otfbm/maps/pull/50/files  Since @g set the opacity to 100% it overrides the h gird lines opacity setting.  This looks like a separate bug.  I can fix this in a separate issue and PR later.

**Comment by Bjorgenas** [comment]
Other known issues or potentially breaking changes currently are:

1.  The border shows some of the image when offset with i.e. c3:10x10 in the url (thanks to Blobulent for reporting this.)
2.   Change in fow default behavior. The default @d or empty options will now default to the h default transparency.  @h100 or @dh100 would be current default replacement.

**Comment by Bjorgenas** [comment]
I have fixed the issue where the FOW was not drawn over the right and bottom grid areas.

File: board.js [hunk 1]
```diff
@@ -411,8 +413,8 @@ module.exports = class Board {
           img.height - offsetY - offsetTrimY,
           this.padding - this.panX * this.gridsize,
           this.padding - this.panY * this.gridsize,
-          (img.width * this.backgroundZoom * this.zoom) - scaledOffsetX - scaledOffsetTrimX,
-          (img.height * this.backgroundZoom * this.zoom) - scaledOffsetY - scaledOffsetTrimY,
+          this.dWitdh,
```
Comment (digitalsadhu) [inline board.js:416]: dWidth

**Review by digitalsadhu — CHANGES_REQUESTED** [review]
Hey thanks so much for this! I've done some testing locally and read the code changes now.

Some thoughts:
I don't think we should change the default behaviour. Doing so risks a whole bunch of maps suddenly revealing things when re roll out.
Given that, I think we shouldn't couple the grid opacity to the FOW opacity. 
A new global setting for FOW opacity is probably the way to go.

How about adding an options `f` to control the FOW transparency?

Eg. `@f50` for 50% opacity?

File: board.js (hunk 1 above)
Comment (Bjorgenas) [reply board.js:416]: 100% agree about decoupling the FOW transparency from the grid transparency and keeping default behavior unchanged.

I will implement f in the options parser similar to h  and add a variable.

Thanks for the feedback! :)

Comment (digitalsadhu) [reply board.js:416]: Great stuff thanks!

**Review by mjs1 — APPROVED** [review]
looks good

**Review by digitalsadhu — APPROVED** [review]
Finally got around to testing this and it looks great!

**Comment by digitalsadhu** [comment]
Just noticed that the following is changed with this PR:
test-integration-image-snapshots test mjs-TAP-FoW---basic---dark-mode diff
test-integration-image-snapshots test mjs-TAP-FoW---basic---light-mode diff

IE. When FOW opacity is 100% then you don't see gridlines any more. Can't decide if this is a bug or a feature.
Thoughts @mjs1 @Bjorgenas ?

**Comment by Bjorgenas** [comment]
What is the default behavior for other VTT?

**Comment by mjs1** [comment]
I don't mind the missing lines, it might be a hassle on big maps when there is lots of fog but why would anyone do that...

**Comment by digitalsadhu** [comment]
I also don't have a strong preference. Shall we merge as is and see how that goes down with the user base?

**Pull request merged**